blockonomics/whmcs-bitcoin-plugin
# Enhance Blockonomics Gateway with Store Management and Improved Test Setup

## Overview
This PR introduces significant improvements to the Blockonomics payment gateway module, focusing on store management integration and a more robust test setup process.

## Key Changes

### Store Management
- Added new store name display functionality that automatically fetches and shows the associated store name from Blockonomics API
- Implemented automatic store name updates when API key changes
- Added new `storesetup.php` endpoint to handle store information retrieval

### Test Setup Improvements
- Redesigned test setup UI with a dedicated button and cleaner response display
- Enhanced error handling with more descriptive messages
- Simplified test results display to a single response area instead of multiple rows
- Added store validation during test setup to ensure proper store configuration

![image](https://github.com/user-attachments/assets/ddef713a-f044-4adc-ae83-d7a9684cee09)


### Code Improvements
- Refactored API communication code for better maintainability
- Added constants for API base URLs
- Improved error handling and messaging throughout
- Removed redundant BCH (Bitcoin Cash) related code
- Enhanced callback URL management

## Testing Instructions
1. Install the module and navigate to the Blockonomics configuration page
2. Enter a valid API key and verify that the store name is automatically populated
3. Test the "Test Setup" functionality with both valid and invalid configurations
4. Verify that error messages are clear and helpful
5. Confirm that callback URLs are properly set up when testing





## Future Work & Improvements Needed

### Installation Wizard
A guided setup wizard would improve first-time user experience:

#### Phase 1: Account & Wallet Setup
- Guide users through creating Blockonomics account
- Help setting up wallet configuration
- Instructions for obtaining API key

#### Phase 2: Store Configuration  
- Check if store exists for API key
- If store exists:
  - Display store details
  - Show success message
- If no store:
  - Guide user to create store on Blockonomics
  - Provide store creation form
  - Auto-configure callback URLs

#### Phase 3: Final Configuration
- Display configured store details
- Show API key status
- Present advanced settings options
- Provide testing tools
